components/button/src/button.tsx

```diff
@@ -6,7 +6,7 @@ import type { WithAsChild } from "../../types/src/as-child";
 import type { VariantProps } from "../../types/src/variant-props";
 
 const buttonVariants = cva(
-	"items-center justify-center rounded-md focus-visible:outline-none focus-visible:ring-4 disabled:pointer-events-none disabled:opacity-50 whitespace-nowrap sm:text-sm",
+	"items-center justify-center rounded-md focus-visible:outline-none focus-visible:ring-4 disabled:pointer-events-none disabled:opacity-50 whitespace-nowrap sm:text-sm aria-[disabled]:pointer-events-none aria-[disabled]:opacity-50",
```

Comment:
we don't actually need the arbitrary variant syntax for these, they're included as part of tailwind!

![image](https://github.com/ngrok-oss/mantle/assets/34115417/86b925e6-87db-4e92-9e5f-183cdf084f47)
https://tailwindcss.com/docs/hover-focus-and-other-states#aria-states

```suggestion
	"items-center justify-center rounded-md focus-visible:outline-none focus-visible:ring-4 disabled:pointer-events-none disabled:opacity-50 whitespace-nowrap sm:text-sm aria-disabled:pointer-events-none aria-disabled:opacity-50",
```